python/ray/serve/controller.py

```diff
@@ -507,6 +497,9 @@ async def __init__(self,
         # at any given time.
         self.write_lock = asyncio.Lock()
 
+        self.long_pull_host = LongPullerHost()
```

Comment:
Don't we need to snapshot these so that we can recover the version IDs on crash?

Reply:
No need to snapshot here because all clients will just retry their `listen_on_changed` call, the host will be initialized with a different `snapshot_ids` and immediately sends the recovered objects over (even though these data can be the same).